tests/Application/Actions/Hooks/StripeChargeUpdateTest.php

```diff
@@ -367,6 +372,64 @@ public function testSuccessfulFullRefund(): void
         $webhookSecret = $container->get('settings')['stripe']['accountWebhookSecret'];
         $time = (string) time();
 
+        $chatterProphecy = $this->prophesize(StripeChatterInterface::class);
+        // Double-check that the normal success case isn't messaging Slack.
+        $chatterProphecy->send(Argument::type(ChatMessage::class))->shouldNotBeCalled();
+
+        $donationRepoProphecy = $this->prophesize(DonationRepository::class);
+        $donationRepoProphecy
+            ->findOneBy(['chargeId' => 'ch_externalId_123'])
+            ->willReturn($donation)
+            ->shouldBeCalledOnce();
+
+        $donationRepoProphecy
+            ->releaseMatchFunds($donation)
+            ->shouldBeCalledOnce();
+
+        $donationRepoProphecy
+            ->push(Argument::type(Donation::class), false)
+            ->willReturn(true)
+            ->shouldBeCalledOnce();
+
+        $entityManagerProphecy = $this->prophesize(EntityManagerInterface::class);
+
+        $container->set(EntityManagerInterface::class, $entityManagerProphecy->reveal());
+        $container->set(DonationRepository::class, $donationRepoProphecy->reveal());
+        $container->set(StripeChatterInterface::class, $chatterProphecy->reveal());
+
+        $request = $this->createRequest('POST', '/hooks/stripe', $body)
+            ->withHeader('Stripe-Signature', $this->generateSignature($time, $body, $webhookSecret));
+
+        $response = $app->handle($request);
+
+        $this->assertEquals('ch_externalId_123', $donation->getChargeId());
+        $this->assertEquals(DonationStatus::Refunded, $donation->getDonationStatus());
+        $this->assertEquals('1.00', $donation->getTipAmount());
+        $this->assertEquals(200, $response->getStatusCode());
+    }
+
+    public function testSuccessfulOverRefund(): void
```

Comment:
`Successful` feels like a bit of a meaninguless word to me. Would it be possible to make the test name a full sentence that describes the main thing we're asserting? Looking at the test I think it might be `testOverRefundSendsSlackNoticeAndUpdatesRecord`. But the fact that there's an `and` in there makes me think maybe it should be two tests.

Reply:
For now I've given it this name and added a comment. I think that releasing match funds is important too – adding this test helped me spot that the logic for that needed to change.